src/test/emulators/functionsEmulatorRuntime.spec.ts

```diff
@@ -233,54 +254,134 @@ describe("FunctionsEmulator-Runtime", () => {
         expect(logs["set-firestore-settings"]).to.eq(1);
       }).timeout(TIMEOUT_MED);
 
-      it("should redirect Firestore write to emulator", async () => {
-        const onRequestCopy = _.cloneDeep(
-          FunctionRuntimeBundles.onRequest
-        ) as FunctionsRuntimeBundle;
+      it("should provide the same stubs through admin-global or through the default app", async () => {
```

Comment:
There are 5 new tests here, 1 old test deleted, and then the rest are just refactored to remove boilerplate.